### PR TITLE
Add more beforematch alternatives

### PR DESCRIPTION
**Pull request by josepharhar**
This adds some alternatives about leveraging the details/summary
elements and changing hidden-matchable to be an attribute instead of a
css property.

### CODE REVIEW

File: explainers/beforematch.md [hunk 1]
```diff
@@ -404,6 +404,58 @@ could be applied to the entire ancestor chain.
...
 
+#### `<details>`/`<summary>` auto-expanding
+Instead of having a generic hidden-matchable property or a specialized
+beforematch event, we could just slightly tweak the `<details>` element to make
+collapsed content searchable. This addresses most of the use cases we have seen
+for the beforematch event, and the details element already has a toggle event
+which would act like the beforematch event. This could be implemented either by
+adding an attribute to the details element saying that the collapsed content
+should be searchable, or just by making it searchable as-is because find-in-page
+is not specced.
```
Comment (vmpstr) [inline explainers/beforematch.md:415]: If you want you can add a code example here to show what the change would be something like

```html
<details searchable>
  <summary>Summary content</summary>
  Detailed content
</details>
```

File: explainers/beforematch.md [hunk 2]
```diff
@@ -404,6 +404,58 @@ could be applied to the entire ancestor chain.
...
+Instead of having `content-visibility: hidden-matchable` and a separate
+`beforematch` event that fires whenever text is searched for, we could have
+a `hidden-matchable` element attribute which would function like the
+`content-visibility: hidden-matchable` css property. By having an element
+attribute instead of a css property, we could have the browser reveal the
+content and scroll on its own by removing the `hidden-matchable` attribute
+instead of needing to have script to do so. We would also only fire
+`beforematch` on the element with the `hidden-matchable` property as soon as it
+is revealed, rather than firing the event on the nearest block-level element
+every time any text is searched for.
```
Comment (vmpstr) [inline explainers/beforematch.md:439]: Likewise here, maybe just a short code example

File: explainers/beforematch.md (hunk 1 above)
Comment (josepharhar) [reply explainers/beforematch.md:415]: Thanks! done

File: explainers/beforematch.md (hunk 2 above)
Comment (josepharhar) [reply explainers/beforematch.md:450]: Thanks! done

**Pull request merged**